### DISCUSSION TITLE
3D animations over an otherwise 2D game

### DISCUSSION THREAD

**Discussion by hhhapz** [enable3d/enable3d#382] [Help / Questions]
Hey, I have a basic 2d video game that I'm working on that uses Phaser. I have a few 3d animations, (e.g. for when collecting rewards mainly) that I would like to be able to play, complete the animation, and then hide. Initially I tried just having a 3d scene that is rendered above my 2d game, but obviously that wont work, because there's no concept of transparency.

I saw #78 which hints that it's possible to render 2d objects in the 3d world. There's a few examples that brought me closer, notably https://enable3d.io/examples/sync-2d-to-3d.html, but this isn't quite what I'd be wanting either, I don't think, since the 2d object is always in front.

I'm not very experienced with the 3d world and its concepts, so I apologize for what might be straightforward concepts, but some guidance in the direction as to how I would go about implementing this would be appreciated.

Is it even possible to make the 2d objects be in the "back"ground, and have gltf animations doing their thing in the "fore"ground?

**Comment by Penumbra69** [comment]
What we're trying to do right now is a bit of an inverse: We're creating a 3D Scene that takes the whole page, and then rendering 2D elements over it with transparency where needed.. That way the scene is always available, and you just need to position your camera/objects and either hide or reduce opacity on your 2D elements to see the 3D.. 

It's not going to render "on top of" like you would with HTML5 particle emitters or something, but it does give you a 2D/3D UI.
